metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ParserRunnerImpl.java

```diff
@@ -0,0 +1,216 @@
+                sensorType));
+      }
+
+      SensorParserConfig parserConfig = parserConfigSupplier.get().getSensorParserConfig(sensorType);
+      // create message parser
+      MessageParser<JSONObject> parser = ReflectionUtils
+              .createInstance(parserConfig.getParserClassName());
+
+      // create message filter
+      MessageFilter<JSONObject> filter = null;
+      parserConfig.getParserConfig().putIfAbsent("stellarContext", stellarContext);
+      if (!StringUtils.isEmpty(parserConfig.getFilterClassName())) {
+        filter = Filters.get(
+                parserConfig.getFilterClassName(),
+                parserConfig.getParserConfig()
+        );
+      }
+      parser.configure(parserConfig.getParserConfig());
+      parser.init();
+      sensorToParserComponentMap.put(sensorType, new ParserComponent(parser, filter));
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  protected Optional<ParserResult> processMessage(String sensorType, JSONObject message, RawMessage rawMessage,
+                                                  MessageParser<JSONObject> parser,
+                                                  ParserConfigurations parserConfigurations) {
+    Optional<ParserResult> parserResult = Optional.empty();
+    SensorParserConfig sensorParserConfig = parserConfigurations.getSensorParserConfig(sensorType);
+    sensorParserConfig.getRawMessageStrategy().mergeMetadata(
+            message,
+            rawMessage.getMetadata(),
+            sensorParserConfig.getMergeMetadata(),
+            sensorParserConfig.getRawMessageStrategyConfig()
+    );
+    message.put(Constants.SENSOR_TYPE, sensorType);
+    applyFieldTransformations(message, rawMessage.getMetadata(), sensorParserConfig);
+    if (!message.containsKey(Constants.GUID)) {
+      message.put(Constants.GUID, UUID.randomUUID().toString());
+    }
+    MessageFilter<JSONObject> filter = sensorToParserComponentMap.get(sensorType).getFilter();
+    if (filter == null || filter.emit(message, stellarContext)) {
+      boolean isInvalid = !parser.validate(message);
+      List<FieldValidator> failedValidators = null;
+      if (!isInvalid) {
+        failedValidators = getFailedValidators(message, parserConfigurations);
+        isInvalid = !failedValidators.isEmpty();
+      }
+      if (isInvalid) {
+        MetronError error = new MetronError()
+                .withErrorType(Constants.ErrorType.PARSER_INVALID)
+                .withSensorType(Collections.singleton(sensorType))
+                .addRawMessage(message);
+        Set<String> errorFields = failedValidators == null ? null : failedValidators.stream()
+                .flatMap(fieldValidator -> fieldValidator.getInput().stream())
+                .collect(Collectors.toSet());
+        if (errorFields != null && !errorFields.isEmpty()) {
+          error.withErrorFields(errorFields);
+        }
+        parserResult = Optional.of(new ParserResult(sensorType, error, rawMessage.getMessage()));
```

Comment:
@merrimanr you and @ottobackwards have landed in a similar place. I like what @ottobackwards has done with keeping the MessageParser interface backwards compatible. There will be some merge conflicts to resolve, but I think the API designs will come together nicely.

Reply:
I am expecting some merge conflicts with https://github.com/apache/metron/pull/1184.  Happy to handle those conflicts if that goes in before this.